addons/product/static/src/js/pricelist_report/product_pricelist_report.js

```diff
@@ -0,0 +1,213 @@
+                return {
+                    displayPricelistTitle: this.displayPricelistTitle,
+                    quantities: this.quantities,
+                    selectedPricelist: this.selectedPricelist,
+              };
+            },
+        });
+    }
+
+    // getters and setters
+
+    get displayPricelistTitle() {
+        return this.state.displayPricelistTitle;
+    }
+
+    get html() {
+        return this.state.html;
+    }
+
+    get pricelists() {
+        return this.state.pricelists;
+    }
+
+    get quantities() {
+        return this.state._quantities;
+    }
+
+    set quantities(value) {
+        this.state._quantities = value;
+    }
+
+    get reportParams() {
+        return {
+            active_model: this.activeModel || 'product.template',
+            active_ids: this.activeIds || [],
+            display_pricelist_title: this.displayPricelistTitle || '',
+            pricelist_id: this.selectedPricelist.id || '',
+            quantities: this.quantities || [1],
+        };
+    }
+
+    get selectedPricelist() {
+        return this.state.selectedPricelist;
+    }
+
+    // orm calls
+
+    getPricelists() {
+        return this.orm.searchRead("product.pricelist", [], ["id", "name"]);
+    }
+
+    async renderHtml() {
+        let html = await this.orm.call(
+            "report.product.report_pricelist", "get_html", [], {data: this.reportParams}
+        );
+        this.state.html = markup(html);
+    }
+
+    // events
+
+    async onClickAddQty(ev) {
+        ev.preventDefault(); // avoid automatic reloading of the page
+
+        if (this.quantities.length >= this.MAX_QTY) {
+            let message = sprintf(
+                this.env._t("At most %s quantities can be displayed simultaneously. Remove a selected quantity to add others."),
+                this.MAX_QTY
+            );
+            await this.action.doAction(sendCustomNotification("warning", message));
+            return;
+        }
+
+        const qty = parseInt($("input.add-quantity-input")[0].value);
+        if (qty && qty > 0) {
+            // Check qty already exist.
+            if (this.quantities.indexOf(qty) === -1) {
+                this.quantities.push(qty);
+                this.quantities = this.quantities.sort((a, b) => a - b);
+                this.renderHtml();
+            } else {
+                let message = sprintf(this.env._t("Quantity already present (%s)."), qty);
+                await this.action.doAction(sendCustomNotification("info", message));
+            }
+        } else {
+            await this.action.doAction(
+                sendCustomNotification("info", this.env._t("Please enter a positive whole number."))
+            );
+        }
+    }
+
+    onClickLink(ev) {
+        ev.preventDefault();
+
+        let classes = ev.target.getAttribute("class", "");
+        let resModel = ev.target.getAttribute("data-model", "");
+        let resId = ev.target.getAttribute("data-res-id", "");
```

Comment:
const

Reply:
I will keep it in mind for the future, indeed these values do not change.